cmd/estuary-shuttle/rpc.go

```diff
@@ -319,61 +321,81 @@ func (s *Shuttle) handleRpcAggregateStagedContent(ctx context.Context, cmd *drpc
 	default:
 		return err
 	case nil:
-		// exists already
+		// if it was aggregated successfully but estuary was not notified,, resend message
+		if p.Active {
+			return s.resendPinComplete(ctx, p)
+		}
 		return nil
 	case gorm.ErrRecordNotFound:
 		// normal case
 	}
 
-	totalSize := int64(len(cmd.ObjData))
 	for _, c := range cmd.Contents {
-		var aggr Pin
-		if err := s.DB.First(&aggr, "content = ?", c).Error; err != nil {
+		var cont Pin
+		if err := s.DB.First(&cont, "content = ?", c.ID).Error; err != nil {
 			// TODO: implies we dont have all the content locally we are being
 			// asked to aggregate, this is an important error to handle
 			return err
 		}
 
-		if !aggr.Active || aggr.Failed {
-			return fmt.Errorf("content i am being asked to aggregate is not pinned: %d", c)
+		if !cont.Active || cont.Failed {
+			return fmt.Errorf("content i am being asked to aggregate is not pinned: %d", c.ID)
 		}
-		totalSize += aggr.Size
 	}
 
-	pin := &Pin{
-		Content:   cmd.DBID,
-		Cid:       util.DbCID{CID: cmd.Root},
-		UserID:    cmd.UserID,
-		Size:      totalSize,
-		Active:    false,
-		Pinning:   true,
-		Aggregate: true,
+	bserv := blockservice.New(s.Node.Blockstore, s.Node.Bitswap)
```

Comment:
seems like we can make a helper to do this to de-dupe with `CreateAggregate` in replication.go

Reply:
yeah, we can do that on a later PR

Reply:
refactoring this part between estuary and shuttle is a bit ticky at the moment